plugin/trino-postgresql/src/test/java/io/trino/plugin/postgresql/TestPostgreSqlConnectorTest.java

```diff
@@ -580,6 +581,13 @@ public void testTimestampColumnAndTimestampWithTimeZoneConstant()
         }
     }
 
+    @Test
+    public void testPruneUnnecessaryGroupingKeysInPushdown() {
```

Comment:
Just my manual test I used to verify the plan did not include the `regionkey` in the pushed down query. Will write proper tests using a mockconnector
```
Query[SELECT min("nationkey") AS "_pfgnrtd_0" FROM "tpch"."nation" GROUP BY "regionkey"]
```

Reply:
Not sure what mockconnector you mean.

We have query building unit tests in TestJdbcQueryBuilder.
We could have end-to-end tests covering query building with a real connector, like PostgreSQL.
We could capture queries on PostgreSQL side (we already have infra for that), or have a test-only functionality capturing queries issued by the connector.

Reply:
I looked at other rule tests (for example: https://github.com/trinodb/trino/blob/master/core/trino-main/src/test/java/io/trino/sql/planner/iterative/rule/TestPushProjectionIntoTableScan.java)

I noticed they create a mock connector such that you can define applyProjection/[in this case applyAggregation] to do whatever you want.

We probably also want an end to end test in BaseConnectorTests, but I figured having a unit test would be helpful too.

Reply:
I added tests:
- Verifying the correct behavior of aggregations without grouping projections in `TestJdbcQueryBuilder`
- Verifying unused grouping columns are pruned from the pushed down query to Postgres in `TestPostgreSqlConnectorTest`. This could be moved to `BaseJdbcConnectorTest`, the tradeoff being we are explicitly asserting against generated SQL and as QueryBuilder evolves that might become brittle.